File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/enums/SqlTypeEnum.java
package org.javamaster.invocationlab.admin.enums;

/**
 * @author yudong
 * @date 2023/9/3
 */
public enum SqlTypeEnum {
    SELECT("select"),
    DELETE("delete"),
    UPDATE("update"),
    INSERT("insert"),
    CREATE("create"),
    DROP("drop"),
    ALTER("alter"),
    ;

    public final String type;

    SqlTypeEnum(String type) {
        this.type = type;
    }

    public static SqlTypeEnum getByType(String type) {
        for (SqlTypeEnum value : SqlTypeEnum.values()) {
            if (value.type.equals(type)) {
                return value;
            }
        }
        throw new IllegalArgumentException(type);
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/enums/SqlTypeEnum.kt
package org.javamaster.invocationlab.admin.enums

/**
 * @author yudong
 * @date 2023/9/3
 */
enum class SqlTypeEnum(@JvmField val type: String) {
    DELETE("delete"),
    UPDATE("update"),
    INSERT("insert"),
    CREATE("create"),
    DROP("drop"),
    ALTER("alter"),
    ;

    companion object {
        @JvmStatic
        fun getByType(type: String): SqlTypeEnum {
            for (value in entries) {
                if (value.type == type) {
                    return value
                }
            }
            throw IllegalArgumentException(type)
        }
    }
}
